src/test/scala/clump/FunctionIdentitySpec.scala

```diff
@@ -0,0 +1,51 @@
+package clump
+
+import org.junit.runner.RunWith
+import org.specs2.runner.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class FunctionIdentitySpec extends Spec {
+
+  def function(string: String, int: Int) = ???
+
+  "detects that the same function created two times is the same" >> {
+    "without parameters from the outer scope" in {
+      def test = function _
+      FunctionIdentity(test) mustEqual FunctionIdentity(test)
+    }
+    "with parameters from the outer scope" in {
+      val string = "test"
+      def test = (int: Int) => function(string, int)
+      FunctionIdentity(test) mustEqual FunctionIdentity(test)
+    }
+  }
+
+  "detects that functions created in different places are different" >> {
```

Comment:
Yeah, it would be better, but it isn't possible to implement using runtime reflection. I was thinking and actually there is another way to implement the function identity by using macros. We can revisit the implementation in the future if this limitation becomes a problem.